app/controllers/edge_controller.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+class EdgeController < RestController
+
+  def all_secrets
+    allowed_params = %i[account limit offset]
+    options = params.permit(*allowed_params)
+      .slice(*allowed_params).to_h.symbolize_keys
+    begin
+      verify_edge_host(options)
+      #scope = Resource.visible_to(current_user).search(options)
+      #scope = Resource.visible_to(current_user).search
+      offset = options[:offset]
+      limit = options[:limit]
+      scope = Resource.where(:resource_id.like("conjur:variable:data%"))
+      scope = scope.order(:resource_id).limit(
+        (limit || 1000).to_i,
+        (offset || 0).to_i
+      )
+    rescue ApplicationController::Forbidden
+      raise
+    rescue ArgumentError => e
+      raise ApplicationController::UnprocessableEntity, e.message
+    end
+    results = []
+    variables = scope.eager(:permissions).eager(:secrets).all
+    variables.each do |variable|
+      variableToReturn = {}
+      variableToReturn[:id] = variable[:resource_id]
+      variableToReturn[:owner] = variable[:owner_id]
+      variableToReturn[:permissions] =  variable.permissions.select{|h| h[:privilege].eql?('execute')}
+      unless variable.last_secret.nil?
+        variableToReturn[:version] = variable.last_secret.version
+        variableToReturn[:value] = variable.last_secret.value
+      end
+      results  << variableToReturn
+    end
+    render(json: results)
+  end
+
+  def all_hosts
+    allowed_params = %i[account limit offset]
+    options = params.permit(*allowed_params)
+                    .slice(*allowed_params).to_h.symbolize_keys
+    verify_edge_host(options)
+    offset = options[:offset]
+    limit = options[:limit]
+    scope = Role.where(:role_id.like("conjur:host:data%"))
```

Comment:
Is there no Exceptions that can be thrown? 

For example from above I saw 

`rescue ApplicationController::Forbidden
      raise
    rescue ArgumentError => e
      raise ApplicationController::UnprocessableEntity, e.message`